pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -84,6 +85,9 @@ def __init__(
         # initialization
         self._device_type = DeviceType.CPU
         self._distrib_type = None
+        self.ipython_compatible_distrib_types = [
+            DistributedType.DP, DistributedType.DDP_SPAWN, DistributedType.DDP_SHARDED_SPAWN
+        ]
```

Comment:
maybe it would be better to move this into the `DistributedType` class as a method that returns the list of incompatible types. 
@carmocca suggested something similar in another place.

Reply:
Thanks for the test! Maybe it would be better to add some tests on the compatible ones to test false negative? The suggestion makes sense. I thought it would be better to make ipython compatibility an attribute of this enum, but a method returning a list seems good enough for now

Reply:
Created #6079 to track this